### PR TITLE
Expand `std::path::Component` documentation.

### PR DESCRIPTION
**Pull request by frewsxcv**
Indicate how it gets created and add an example.

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -525,6 +525,26 @@ impl<'a> Hash for PrefixComponent<'a> {
 ///
 /// See the module documentation for an in-depth explanation of components and
 /// their role in the API.
+///
+/// This `struct` is created from iterating over the [`path::Components`]
```
Comment (GuillaumeGomez) [inline src/libstd/path.rs]: I'm not sure to understand in here. You're talking about the `Components` struct but this is supposed to be the doc for the `Component` enum. Any further explanation please? :)

Comment (frewsxcv) [inline src/libstd/path.rs]: The only way (that I know) to create a `Component` `struct` is by iterating over the `Components` `struct`.

Comment (frewsxcv) [inline src/libstd/path.rs]: Nevermind. I'll change this to `enum` later.

Comment (GuillaumeGomez) [inline src/libstd/path.rs]: So it was just the wrong word. Ping me once changed then. :stuck_out_tongue:

Comment (frewsxcv) [inline src/libstd/path.rs]: Ping.

**Comment by GuillaumeGomez** [comment]
All good, thanks!

@bors: r+ rollup

**Pull request merged**